backend/src/apiserver/server/experiment_server.go

```diff
@@ -328,26 +328,23 @@ func (s *ExperimentServer) canAccessExperiment(ctx context.Context, experimentID
 		// Skip authorization if not multi-user mode.
 		return nil
 	}
-
+	namespace := ""
 	if len(experimentID) > 0 {
 		experiment, err := s.resourceManager.GetExperiment(experimentID)
 		if err != nil {
 			return util.Wrap(err, "Failed to authorize with the experiment ID")
 		}
-		if len(resourceAttributes.Namespace) == 0 {
-			if len(experiment.Namespace) == 0 {
-				return util.NewInternalServerError(
-					errors.New("Empty namespace"),
-					"The experiment doesn't have a valid namespace",
-				)
-			}
-			resourceAttributes.Namespace = experiment.Namespace
-		}
+		namespace = experiment.Namespace
 		if len(resourceAttributes.Name) == 0 {
 			resourceAttributes.Name = experiment.Name
 		}
 	}
-
+	if s.resourceManager.IsEmptyNamespace(resourceAttributes.Namespace) {
+		if len(namespace) == 0 {
```

Comment:
Since `namespace` is a string, maybe we can just compare it with an empty string instead?

Reply:
I agree that both are okay and in fact they get compiled to the same result. I do support @Linchin's comment as it is probably more explicit.